native/core/src/events/news.rs

```diff
@@ -30,61 +30,142 @@ pub enum NewsContent {
     pub fn type_str(&self) -> String {
         match self {
-            NewsContent::Audio(_) => "audio".to_owned(),
-            NewsContent::File(_) => "file".to_owned(),
-            NewsContent::Image(_) => "image".to_owned(),
-            NewsContent::Location(_) => "location".to_owned(),
-            NewsContent::Text(_) => "text".to_owned(),
-            NewsContent::Video(_) => "video".to_owned(),
+            FallbackNewsContent::Audio(_) => "audio".to_owned(),
+            FallbackNewsContent::File(_) => "file".to_owned(),
+            FallbackNewsContent::Image(_) => "image".to_owned(),
+            FallbackNewsContent::Location(_) => "location".to_owned(),
+            FallbackNewsContent::Text(_) => "text".to_owned(),
+            FallbackNewsContent::Video(_) => "video".to_owned(),
         }
     }
 
     pub fn audio(&self) -> Option<AudioMessageEventContent> {
         match self {
-            NewsContent::Audio(content) => Some(content.clone()),
+            FallbackNewsContent::Audio(content) => Some(content.clone()),
             _ => None,
         }
     }
 
     pub fn file(&self) -> Option<FileMessageEventContent> {
         match self {
-            NewsContent::File(content) => Some(content.clone()),
+            FallbackNewsContent::File(content) => Some(content.clone()),
             _ => None,
         }
     }
 
     pub fn image(&self) -> Option<ImageMessageEventContent> {
         match self {
-            NewsContent::Image(content) => Some(content.clone()),
+            FallbackNewsContent::Image(content) => Some(content.clone()),
             _ => None,
         }
     }
 
     pub fn location(&self) -> Option<LocationMessageEventContent> {
         match self {
-            NewsContent::Location(content) => Some(content.clone()),
+            FallbackNewsContent::Location(content) => Some(content.clone()),
             _ => None,
         }
     }
 
     pub fn text(&self) -> Option<TextMessageEventContent> {
         match self {
-            NewsContent::Text(content) => Some(content.clone()),
+            FallbackNewsContent::Text(content) => Some(content.clone()),
             _ => None,
         }
     }
 
     pub fn video(&self) -> Option<VideoMessageEventContent> {
         match self {
-            NewsContent::Video(content) => Some(content.clone()),
+            FallbackNewsContent::Video(content) => Some(content.clone()),
             _ => None,
         }
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "type")]
+pub enum NewsContent {
+    /// An image message.
+    Image(ImageMessageEventContent),
+    /// A video message.
+    Video(VideoMessageEventContent),
+    /// A text message.
+    Text(TextMessageEventContent),
+    /// An audio message.
+    Audio(AudioMessageEventContent),
+    /// A file message
+    File(FileMessageEventContent),
+    /// A location message.
+    Location(LocationMessageEventContent),
+
+    #[serde(untagged)]
```

Comment:
Let's add some docs here:
```suggestion
    /// Backwards-compatible fallback support for previous untagged version
    /// only for reading existing events.
    #[serde(untagged)]
```

Reply:
Forgot it since you mentioned. Let me add!